packages/flutter/lib/src/widgets/animated_size.dart

```diff
@@ -0,0 +1,59 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/rendering.dart';
+import 'package:meta/meta.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+
+/// Animated widget that automatically transitions its size over a given
+/// duration whenever the given child's size changes.
+class AnimatedSize extends SingleChildRenderObjectWidget {
+  /// Creates a widget that animates its size to match that of its child.
+  ///
+  /// The [curve] and [duration] arguments must not be null.
+  AnimatedSize({
+    Key key,
+    Widget child,
+    this.alignment: FractionalOffset.center,
+    this.curve: Curves.linear,
+    @required this.duration
+  }) : super(key: key, child: child);
+
+  /// How to align the child.
+  ///
+  /// The x and y values of the alignment control the horizontal and vertical
+  /// alignment, respectively. An x value of 0.0 means that the left edge of
+  /// the child is aligned with the left edge of the parent whereas an x value
+  /// of 1.0 means that the right edge of the child is aligned with the right
+  /// edge of the parent. Other values interpolate (and extrapolate) linearly.
+  /// For example, a value of 0.5 means that the center of the child is aligned
+  /// with the center of the parent.
+  final FractionalOffset alignment;
+
+  /// The curve to apply when animating the parameters of this container.
```

Comment:
No need to be generic here. Maybe: the animation curve when transitioning this widget's size to match the child's size.